core-rust/core-api-server/src/core_api/browse.rs

```diff
@@ -186,6 +194,100 @@ impl ObjectFieldMeta {
     }
 }
 
+/// A lister of Engine's Nodes.
+pub struct EngineNodeLister<'s, S> {
+    database: &'s S,
+}
+
+impl<'s, S: QueryableProofStore + ReadableTreeStore> EngineNodeLister<'s, S> {
```

Comment:
_(note from author)_

This StateHashTree-based impl may be regarded as temporary, if we want to migrate to NodeAncestryRecord-based impl: see discussion at https://radixdlt.atlassian.net/wiki/spaces/RPNV1/pages/3189211214/RNP-12+-+Browse+API+sub-API+within+Core+API#Implementation-details

Still, I consider it good-enough to be merged in v1 of this API.